multiversx/mx-chain-go
# MX-13777: added round in hyperblock transactions

## Reasoning behind the pull request
- previously, the `Round` field of the hyperblocks' transactions was not populated
  
## Proposed changes
- use the round of the block that contained the transaction(s)

## Testing procedure
- start a testnet with nodes using this branch, and perform a GET Rest API call to `<proxy>/hyperblock/by-nonce/<nonce>` (where `<nonce>` is a metablock nonce where transactions were notarized). The `round` field of the transactions should be populated

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?
